GSoC25/NEF/NEF.py

```
@@ -332,46 +330,53 @@ def disambiguate_triple(
+            s_idx = data.subject_index
+            o_idx = data.object_index
+
+            # Validate predicate
+            if pred_uri not in allowed:
+                pred_uri = None      ## if none
 
-        # Validate predicate
-        pred_uri = data.get("predicate", "")
-        if pred_uri not in allowed:
-            pred_uri = allowed[0]
+        except Exception as e:
```

Ruff (0.14.14)

Do not catch blind exception: `Exception`

(BLE001)